pyvista/utilities/fileio.py

```diff
@@ -274,37 +272,35 @@ def read_meshio(filename, file_format = None):
         vtk_type = meshio_to_vtk_type[k]
         numnodes = vtk_type_to_numnodes[vtk_type]
         offset += [next_offset+i*(numnodes+1) for i in range(len(v))]
-        cells.append(numpy.hstack((numpy.full((len(v), 1), numnodes), v)).ravel())
+        cells.append(np.hstack((np.full((len(v), 1), numnodes), v)).ravel())
         cell_type += [vtk_type] * len(v)
         next_offset = offset[-1] + numnodes + 1
 
         # Extract cell data
         if k in mesh.cell_data.keys():
             for kk, vv in mesh.cell_data[k].items():
                 if kk in cell_data:
-                    cell_data[kk] = numpy.concatenate((cell_data[kk], vv))
+                    cell_data[kk] = np.concatenate((cell_data[kk], vv))
                 else:
                     cell_data[kk] = vv
 
     # Create pyvista.UnstructuredGrid object
+    points = mesh.points
+    if points.shape[1] == 2:
+        points = np.c_[points, np.zeros(len(points))]
+
     grid = pyvista.UnstructuredGrid(
-        numpy.array(offset),
-        numpy.concatenate(cells),
-        numpy.array(cell_type),
-        numpy.array(mesh.points, numpy.float64),
+        np.array(offset),
+        np.concatenate(cells),
+        np.array(cell_type),
+        np.array(mesh.points, np.float64),
     )
 
     # Set point data
-    for k, v in mesh.point_data.items():
-        data = vtk.util.numpy_support.numpy_to_vtk(numpy.array(v, numpy.float64))
-        data.SetName(k)
-        grid.GetPointData().AddArray(data)
-
+    grid.point_arrays.update(mesh.point_data)
     # Set cell data
-    for k, v in cell_data.items():
-        data = vtk.util.numpy_support.numpy_to_vtk(numpy.array(v, numpy.float64))
-        data.SetName(k)
-        grid.GetCellData().AddArray(data)
+    grid.cell_arrays.update(cell_data)
+
```

Comment:
FYI, @keurfonluu - PyVista manages the data attributes as dictionaries just like `meshio`

Reply:
``meshio`` point and cell data should be converted to ``np.array(arr, np.float64)`` (or ``np.float32``). I guess there is some kind of conversion that is carried in ``numpy_to_vtk`` depending on the type of the values of the input array.